### DISCUSSION TITLE
npm run dev and npm run build are empty strings now?

### DISCUSSION THREAD

**Discussion by pamelafox** [cookiecutter/cookiecutter-django#4200] [Q&A] [answered]
I'm just syncing my fork to the latest changes from the repo, and I see that package.json changed a fair bit with the webpack change. Specifically, this change happened:

```
   "scripts": {
-    "dev": "gulp",
-    "build": "gulp generate-assets"
+    "dev": "",
+    "build": ""
   }
 }
```

When I make that change locally and call `npm run dev`, it simply does nothing (which is what I'd expect, looking at it). Am I missing some magic by which it'd know which gulp commands to run?
(I set mine up to use Gulp).
Thanks!

**Comment by browniebroke** [comment, marked as answer]
Our `package.json` in the template is a valid JSON file which contains all dependencies (Gulp + Webpack). This is to be able to receive Dependabot updates, which -if I'm not mistaken- don't work with Jinja tags.

We programmatically update the `package.json` file in [the post-gen hooks](https://github.com/cookiecutter/cookiecutter-django/blob/e6014675d276396d20c9fd6b836ab0d4e827292d/hooks/post_gen_project.py#L139) to remove the dependencies which aren't needed and set the npm scripts:

- Gulp: 

  https://github.com/cookiecutter/cookiecutter-django/blob/e6014675d276396d20c9fd6b836ab0d4e827292d/hooks/post_gen_project.py#L159-L162

- Webpack:
	https://github.com/cookiecutter/cookiecutter-django/blob/e6014675d276396d20c9fd6b836ab0d4e827292d/hooks/post_gen_project.py#L166-L169

**Comment by pamelafox** [comment]
Ah, that makes sense! I skipped over post_gen_project when I was looking through the Webpack PR as I didn't realize template changes happened there too. Sorry for making you explain!
